### PR TITLE
Ports EXCLUDE to resolved plan

### PR DESCRIPTION
**Pull request by RCHowell**
## Relevant Issues
https://github.com/partiql/partiql-lang-kotlin/pull/1226

## Description

This is a port of Alan's work on EXCLUDE  (#1226) to the resolve logical plan. 

## Other Information

This passes all existing EXCLUDES tests.

- Any backward-incompatible changes? **[YES/NO]**
No

- Any new external dependencies? **[YES/NO]**
No

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES/NO]**
Yes

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt [hunk 1]
```diff
@@ -917,4 +969,25 @@ internal class PlanTyper(
...
+            }
+        }
+        if (!matchedRoot) handleUnresolvedExcludeRoot(item.root.symbol)
+        return output
+    }
+
+    private fun Identifier.Symbol.isEquivalentTo(other: String): Boolean = when (caseSensitivity) {
+        Identifier.CaseSensitivity.SENSITIVE -> symbol.equals(other)
+        Identifier.CaseSensitivity.INSENSITIVE -> symbol.equals(other, ignoreCase = true)
+    }
```
Comment (alancai98) [inline partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt:992]: Any reason to have `Identifier.Symbol.isEquivalentTo(other: String)` in both this file and `TypeUtils.kt`?

**Review by alancai98 — COMMENTED** [review]
Looks good! I like the factoring approach you took, splitting the logic between `PlanTyper` and `TypeUtils`. Only had a couple minor comments.

File: partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt (hunk 1 above)
Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt:992]: This extension could be public in `org.partiql.plan` actually. But I put it here to keep it private

Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt:992]: I don't mind minor duplication of logic when it's just syntax sugar and lets use keep things private. Internal visibility in Kotlin is Java public

**Review by alancai98 — APPROVED** [review]
🚢 . looks to have a ktlint failure: https://github.com/partiql/partiql-lang-kotlin/actions/runs/6462383725/job/17543877945?pr=1243

**Pull request merged**